sim/core/movement.go

```diff
@@ -0,0 +1,139 @@
+package core
+
+import (
+	"math"
+	"time"
+
+	"github.com/wowsims/cata/sim/core/proto"
+)
+
+type MovementAction struct {
+	PendingAction
+	srcPosition float64       // starting position
+	startTime   time.Duration // starting time of the movement
+	speed       float64       // theoretical movement speed, can be 0
+}
+
+func (action *MovementAction) GetCurrentPosition(sim *Simulation) float64 {
+	return action.srcPosition + float64(sim.CurrentTime-action.startTime)*action.speed/float64(time.Second)
+}
+
+func (unit *Unit) initMovement() {
+	unit.moveAura = unit.GetOrRegisterAura(Aura{
+		Label:     "Movement",
+		ActionID:  ActionID{OtherID: proto.OtherAction_OtherActionMove},
+		Duration:  NeverExpires,
+		MaxStacks: 30,
+
+		OnGain: func(aura *Aura, sim *Simulation) {
+			unit.Moving = true
+		},
+		OnExpire: func(aura *Aura, sim *Simulation) {
+			unit.Moving = false
+		},
+	})
+
+	unit.moveSpell = unit.GetOrRegisterSpell(SpellConfig{
+		ActionID: ActionID{OtherID: proto.OtherAction_OtherActionMove},
+		Flags:    SpellFlagMeleeMetrics,
+
+		ApplyEffects: func(sim *Simulation, target *Unit, spell *Spell) {
+			unit.moveAura.Activate(sim)
+			unit.moveAura.SetStacks(sim, max(int32(unit.DistanceFromTarget), 1))
+		},
+	})
+
+	unit.RegisterResetEffect(func(s *Simulation) {
+		unit.Moving = false
+		unit.movementAction = nil
+	})
```

Comment:
You can make this be part of the moveAura OnExpire maybe so we dont add extra reset effects?